xbmc/video/VideoDatabase.cpp

```diff
@@ -8708,8 +8723,9 @@ std::vector<int> CVideoDatabase::CleanMediaType(const std::string &mediaType, co
                !sourcePathsDeleteDecision->second.second)
         del = false;
 
+      // delete parents folder (and all its items) only, if file needs to be deleted and folder does not exist
       if (scanSettings.parent_name)
-        pathsDeleteDecisions.insert(std::make_pair(m_pDS2->fv(2).get_asInt(), del));
+        pathsDeleteDecisions.insert(std::make_pair(m_pDS2->fv(2).get_asInt(), !CDirectory::Exists( parentPath, false ) && del));
```

Comment:
I needed more than an hour to understand how this code actual work (2 months ago). And for my scenario the code does not work at all (even worse: it deletes half my library every time I delete some movies). 
In my case the variable "del" decides that an entire path should be deleted, without checking if this is necessary (if path exists or contains movies that exist) or even wanted/desired from the user. So, there is something missing...

I guess some test cases would be very helpful for this part of code...
Test: Create Folder structure:
**Movies** (root folder, that should be added to the video library in kodi)
- **HD**
-- **2000** (into this folder add like 3 movies)
Then update database (3 movies found), then delete one of them on the hdd, execute a library clean and at the end you will see, that all 3 movies were deleted from kodi (instead of the one only).

My current workaround: 
Instead of adding **Movies** to the library, I added all quality level individual: **HD, SD, 4K** etc.
Because the folder level is now decreased by one, a lib clean works.

This issue needs to be fixed anyway someday. Because I have very little time atm, I will look into it again next months, so we can close this issue until Kodi 18 release hopefully ;-)